x-pack/filebeat/input/awss3/metrics.go

```diff
@@ -40,10 +42,21 @@ func (m *inputMetrics) Close() {
 	m.unregister()
 }
 
+func (m *inputMetrics) setSQSMessagesWaiting(count int64) {
+	if count == -1 && m.sqsMessagesWaiting == nil {
+		// if metric not initialized, and count is -1, do nothing
+		return
+	}
+
+	m.sqsMessagesWaiting = monitoring.NewInt(m.registry, "sqs_messages_waiting_gauge")
```

Comment:
Should this only happen if `m.sqsMessagesWaiting` is nil?

Reply:
Yep, missed the save

Updated